extension/agenthealth/handler/stats/provider/flag.go

```diff
@@ -17,20 +17,15 @@ const (
 	flagGetInterval = 5 * time.Minute
 )
 
-type BoolFlag int
+type Flag int
 
 const (
-	FlagIMDSFallbackSucceed BoolFlag = iota
+	FlagIMDSFallbackSucceed Flag = iota
 	FlagSharedConfigFallback
 	FlagAppSignal
 	FlagEnhancedContainerInsights
 	FlagRunningInContainer
-)
-
-type StringFlag int
-
-const (
-	FlagMode StringFlag = iota
+	FlagMode
```

Comment:
What is FlagMode used for?

Reply:
It tracks the agent mode: OnPrem, EC2, WithIRSA. Was added as part of https://github.com/aws/amazon-cloudwatch-agent/pull/913 and moved into agenthealth as part of https://github.com/aws/amazon-cloudwatch-agent/pull/925.